src/Console/Command/NewVersionChecker.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Command;
+
+/**
+ * @internal
+ */
+final class NewVersionChecker
+{
+    /**
+     * @var int[]
+     */
+    private $currentVersion;
+
+    /**
+     * @var null|array<int[]>
+     */
+    private $availableVersions;
+
+    /**
+     * @param string $currentVersion
+     */
+    public function __construct($currentVersion)
+    {
+        $this->currentVersion = $this->parseVersion($currentVersion);
```

Comment:
ref https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3010/files#r136204306

`currentVersion` might be `2.0.0-RC`, here, you will assign null to `currentVersion` which claims to be `int[]`, and then you will provide null to array typehinted functions. not gonna work

you have to add releases like `2.0.0-RC` to tests

Reply:
Replaced with `composer/semver` usage.